src/arch/x86/64/c_traps.c

```diff
@@ -39,84 +39,210 @@ static void NORETURN restore_vmx(void)
     /* Do not support breakpoints in VMs, so just disable all breakpoints */
     loadAllDisabledBreakpointState(cur_thread);
 #endif
+    vcpu_restore_guest_msrs(cur_thread->tcbArch.tcbVCPU);
     if (cur_thread->tcbArch.tcbVCPU->launched) {
         /* attempt to do a vmresume */
         asm volatile(
-            // Set our stack pointer to the top of the tcb so we can efficiently pop
-            "movq %[reg], %%rsp\n"
+            TODO this needs to be updated to remain compatible with existing configurations.
+            // Arguments are getting stored in general purpose registers that need to be used.
```

Comment:
Is the `TODO` line missing a comment marker?

In either case, the code is not a got place for TODO items. This should either be resolved or be made an issue.

Reply:
This is a TODO for this PR, I didn't ifdef that part of the ASM yet so I didn't want this to be able to compile until it has been ifdef'd

Reply:
Ah, that explains it. @abrandnewusername if you're planning to help with this PR, this is another place to look at.